internal/manifests/collector/container.go

```diff
@@ -45,6 +45,10 @@ func Container(cfg config.Config, logger logr.Logger, agent v1alpha1.AmazonCloud
 
 	if addConfig {
 		volumeMounts = append(volumeMounts, getVolumeMounts(agent.Spec.NodeSelector["kubernetes.io/os"]))
+
+		if len(agent.Spec.Prometheus) > 0 {
```

Comment:
Is this placed correctly?

Reply:
This is because prometheus resources are created with collector resources, which aligns with what our docs do (https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights-Prometheus-Setup.html) and also allows for the prometheus config to be accessible without the `target-allocator` if that's not enabled (assuming the alternative is to build on the `target-allocator` container.